tests/mir-opt/pre-codegen/clone_as_copy.rs

```diff
@@ -0,0 +1,28 @@
+//@ compile-flags: -Cdebuginfo=full
+
+// Check if we have transformed the nested clone to the copy in the complete pipeline.
+
+#[derive(Clone)]
+struct AllCopy {
+    a: i32,
+    b: u64,
+    c: [i8; 3],
+}
+
+#[derive(Clone)]
+struct CopyLikeStruct {
+    a: i32,
+    b: AllCopy,
+    c: [i8; 3],
+}
+
+// EMIT_MIR clone_as_copy.clone_as_copy.PreCodegen.after.mir
+#[inline(never)]
+fn clone_as_copy(v: &CopyLikeStruct) -> CopyLikeStruct {
+    // CHECK-LABEL: fn clone_as_copy(
+    // CHECK-NOT: = AllCopy { {{.*}} };
+    // CHECK-NOT: = CopyLikeStruct { {{.*}} };
+    // CHECK: _0 = (*_1);
+    // CHECK: return;
+    v.clone()
+}
```

Comment:
Could you add the NestCopy case here too?

Reply:
I think this is already nested. I updated the name.

Reply:
Now that we support enums, could you add a case with an enum?

Reply:
Updated, and left a FIXME.